splunk_topology/test_splunk_topology.py

```diff
@@ -180,14 +180,28 @@ def test_not_dispatch_sids_checks(self, mocked_splunk_helper):
             '_auth_session': _mocked_auth_session
         }, force_reload=True)
 
-        second_persistent_data = self.check._status().data.get(instance.get('url'))
+        second_persistent_data = self.check._status().data.get(instance.get('url')+"components")
+        # The second run_check will finalize the previous saved search id and create a new one,
+        # so we make sure this is the case
+        self.assertEqual(first_persistent_data, second_persistent_data)
 
-        # clear the persistent data created
-        self.check.update_persistent_status(instance.get('url'), None, None, 'clear')
+        # mock the splunkhelper finalize call
+        mocked_splunk_helper.return_value.finalize_sid = mock.MagicMock(side_effect=FinalizeException(None, "Error"))
 
-        # The second run_check will finalize the previous saved search ids and create a new one,
-        # so we make sure this is the case
-        self.assertNotEqual(first_persistent_data[0][1], second_persistent_data[0][1])
+        thrown = False
+        try:
+            self.run_check(config, mocks={
+                '_search': _mocked_search,
+                '_saved_searches': _mocked_saved_searches,
+                '_auth_session': _mocked_auth_session
+            }, force_reload=True)
+        except CheckException:
+            thrown = True
+
+        self.assertTrue(thrown)
+        self.assertEquals(self.service_checks[0]['status'], 2, "service check should have status AgentCheck.CRITICAL")
+        # clear the persistent data created
+        self.check.update_persistent_status(instance.get('url'), "components", None, 'clear')
```

Comment:
why are you clearing the state of the check from here ?

Reply:
If I don't clear the state, then it still exists in the system with key and value. So just to make sure there is no data previously on the system.

Reply:
ah ok, in tests we generally have before and after function to setup and cleanup the SUT (system under test)
do we have something like that here ?

Reply:
no we don't have.

Reply:
oh man, ok lets fix this somewhere else

Reply:
can you please extract a function `teard_down`